js/libs/quantimodo-api.js

```diff
@@ -0,0 +1,424 @@
+            POST('share', ['type', 'inputVariable', 'outputVariable'], measurements, f);
+        },
+
+        connectorsInterface: function (baseURL, defaultConnector) {
+            this.params = {
+                baseURL: typeof baseURL == 'undefined' ? hostUrl : hostUrl + baseURL,
+                connector: typeof defaultConnector == 'undefined' ? null : defaultConnector
+            };
+            this.connector = function (name) {
+                this.params.connector = name;
+                return this;
+            };
+            this.do = function () {
+                var action = arguments[0],
+                    params = {},
+                    f = undefined;
+
+                if (typeof arguments[1] == 'object') {
+                    params = arguments[1];
+                    f = arguments[2];
+                } else if (typeof arguments[1] == 'function') {
+                    f = arguments[1];
+                }
+                switch (action) {
+                    case 'connect':
+                        this.sendRequest('connectors/' + this.params.connector + '/connect', params, f);
+                        break;
+                    case 'disconnect':
+                        this.sendRequest('connectors/' + this.params.connector + '/disconnect', params, f);
+                        break;
+                    case 'update':
+                        this.sendRequest('connectors/' + this.params.connector + '/update', params, f);
+                        break;
+                    case 'info':
+                        this.sendRequest('connectors/' + this.params.connector + '/info', params, f);
+                        break;
+                }
+            };
+            this.listConnectors = function (f) {
+                this.sendRequest('connectors/list', {}, f);
+            };
+            this.sendRequest = function (url, params, f) {
+                console.debug('API Call via QM JS SDK ');
+
+                var that = this;
+                jQuery.ajax(this.params.baseURL + url, {
+                    beforeSend: function (xhr) {
+                        xhr.setRequestHeader('Authorization', 'Bearer ' + accessToken);
+                        if (typeof mashapeKey !== 'undefined' && mashapeKey) {
+                            xhr.setRequestHeader('X-Mashape-Key', mashapeKey);
+                        }
+                    },
+                    data: params,
+                    dataType: 'json'
+                }).done(function (data) {
+                    if (typeof f != 'undefined') {
+                        f(data, that.params.connector);
+                    }
+                });
+
+            };
+        },
+
+        getCurrentUser: function (f) {
+            GET('user/me', ['id', 'wpId', 'displayName', 'loginName', 'email', 'token', 'clientId', 'userRegistered'],
+                null, f, true);
+        },
+
+        url: hostUrl
+    };
+}();
```

Comment:
Wrap an immediate function invocation in parens to assist the reader in understanding that the expression is the result of a function, and not the function itself.